File: worker/metrics.py
from prometheus_client import Counter, start_http_server


processed_total = Counter(
    "video_processed_total",
    "Total de vídeos processados com sucesso",
)


def start_metrics_server():
    start_http_server(8000)
    print("📊 [WORKER] Métricas Prometheus disponíveis em :8000/metrics")

File: src/utils/metrics.go
package utils

import "github.com/prometheus/client_golang/prometheus"

var (
	UploadsTotal = prometheus.NewCounter(
		prometheus.CounterOpts{
			Name: "video_uploads_total",
			Help: "Total de uploads recebidos",
		},
	)

	ProcessingErrorsTotal = prometheus.NewCounter(
		prometheus.CounterOpts{
			Name: "video_processing_errors_total",
			Help: "Total de erros durante o processamento",
		},
	)

	ProcessedTotal = prometheus.NewCounter(
		prometheus.CounterOpts{
			Name: "video_processed_total",
			Help: "Total de vídeos processados com sucesso",
		},
	)
)

func RegisterMetrics() {
	prometheus.MustRegister(UploadsTotal)
	prometheus.MustRegister(ProcessingErrorsTotal)
	prometheus.MustRegister(ProcessedTotal)
}
